core/Pos/Core/Timestamp.hs

```diff
@@ -45,3 +47,9 @@ timestampF = build
 -- Get the current time as a timestamp
 getCurrentTimestamp :: IO Timestamp
 getCurrentTimestamp = Timestamp . round . (*1000000) <$> getPOSIXTime
+
+diffTimestamp :: Timestamp -> Timestamp -> Microsecond
+diffTimestamp t1 t2 = getTimestamp t1 - getTimestamp t2
+
+addMicrosecondsToTimestamp :: Microsecond -> Timestamp -> Timestamp
```

Comment:
Ah! I didn't realise that it was even possible to add two timestamps. Not sure how much sense it makes though. What would the unit of the result be? It's probably clearer and more future proof to do it this way.